File: src/game/tools/rpc/output/wrap/GuildLogWrap.go
package wraps

import (
	rpc "tipcat.com/common/rpc/simple"
)

type GuildLogWrap struct {
	
	Opcode int16 // 操作类型
	CreateTime int64 // 时间
	MemberID int64 // 操作者ID
	MemberName string // 操作者名字
	TargetID int64 // 目标ID
	TargetName string // 目标名字
	GuildRank int16 // 操作官职
	rpc.Wrapper
}

func (w *GuildLogWrap)Decode(pck *rpc.Packet) rpc.Wrapper {
	
	w.Opcode = pck.PopInt16(); 
	w.CreateTime = pck.PopInt64(); 
	w.MemberID = pck.PopInt64(); 
	w.MemberName = pck.PopString(); 
	w.TargetID = pck.PopInt64(); 
	w.TargetName = pck.PopString(); 
	w.GuildRank = pck.PopInt16(); 
	return w
}

func (w *GuildLogWrap)Encode(pck *rpc.Packet) {
	
	pck.PutInt16(w.Opcode) 
	pck.PutInt64(w.CreateTime) 
	pck.PutInt64(w.MemberID) 
	pck.PutString(w.MemberName) 
	pck.PutInt64(w.TargetID) 
	pck.PutString(w.TargetName) 
	pck.PutInt16(w.GuildRank) 
}

File: src/game/tools/rpc/output/wrap/GuildLogWrap.cs
using System;
using Common.Net.Simple;

namespace Game 
{
	public class GuildLogWrap : IWrapper
	{	
		public short Opcode; // 操作类型
		public long CreateTime; // 时间
		public long MemberID; // 操作者ID
		public string MemberName; // 操作者名字
		public long TargetID; // 目标ID
		public string TargetName; // 目标名字
		public short GuildRank; // 操作官职
		
		public void Decode(Packet pck)
		{	
			Opcode = pck.GetShort(); 
			CreateTime = pck.GetLong(); 
			MemberID = pck.GetLong(); 
			MemberName = pck.GetString(); 
			TargetID = pck.GetLong(); 
			TargetName = pck.GetString(); 
			GuildRank = pck.GetShort(); 
		}
		
        public void Encode(Packet pck)
        {	
        	pck.PutShort(Opcode); 
        	pck.PutLong(CreateTime); 
        	pck.PutLong(MemberID); 
        	pck.PutString(MemberName); 
        	pck.PutLong(TargetID); 
        	pck.PutString(TargetName); 
        	pck.PutShort(GuildRank); 
        }
	}
}